core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/FileSourceConfig.java

```diff
@@ -149,28 +149,49 @@ protected void validate() {
       (format.equalsIgnoreCase("avro")) || format.equalsIgnoreCase("parquet"))) {
       throw new IllegalArgumentException("Format can only be 'text', 'avro' or 'parquet'");
     }
+    if (format.equalsIgnoreCase("text") && schema == null) {
```

Comment:
@albertshau This has been added for the case where the schema is not provided and format is text. Noticed this when all the other tests started failing where the schema was not specified.